### PR TITLE
Prevent addition of duplicate plugin dirs to PATH

### PR DESCRIPTION
**Pull request by PamelaM**
### Description
- [x ] Here are some details about my PR

`libexec/pyenv` adds the `.../plugins/*/bin` directories to the PATH. This leads to duplicated PATH entries when `bin_path` is the same as `PYENV_ROOT`.

This PR simply checks if `bin_path` != `PYENV_ROOT` before adding the bin_path `plugins/*/bin` directories to PATH

### Tests
- [ ] My PR adds the following unit tests (if any)

Make sure you have checked all steps below.

### Prerequisite
* [x] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
  * _This is a trivial bug in libexe/pyenv and, therefore, not amenable to either a hook script or a plugin_
* [x] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
  *  _I checked rbenv and this code is specific to a difference between rbenv and pyenv, specifically: pyenv adds the bin_path plugins/*/bin directories, while rbenv does not_

* [x] My PR addresses the following pyenv issue (if any)
  - Closes https://github.com/pyenv/pyenv/issues/XXXX
  - _No existing pyenv issue_

### CODE REVIEW

**Comment by native-api** [comment]
> when `bin_path` is the same as `PYENV_ROOT`

How can this happen?
The only way I see is if you add `pyenv/libexec` to `PATH` instead of `pyenv/bin`. That is not a supported configuration.

**Comment by native-api** [comment]
> when `bin_path` is the same as `PYENV_ROOT`

How can this happen?
`bin_path` (the location of `pyenv` executable) is supposed to be either `$PYENV_ROOT/bin` (git checkout installation), or outside `$PYENV_ROOT` (Homebrew or other package manager installation).

**Comment by PamelaM** [comment]
> > when `bin_path` is the same as `PYENV_ROOT`
> 
> How can this happen?
> `bin_path` (the location of the `pyenv` executable) is supposed to be either `$PYENV_ROOT/bin` (git checkout installation), or outside `$PYENV_ROOT` (Homebrew or other package manager installation).

This happened in a Docker container that pre-installed pyenv and set `ENV PYENV_ROOT=...`

Now that I think about it from outside my tunnel vision, I can see that that's not an good plan, and will have a conversation with that Dockerfile maintainer.

I think the fix is still "safe", but it does add complexity, so it's your call whether to include this or not.

**Review by native-api — APPROVED** [review]
Since the code performs the same operation twice for 2 paths, I guess it makes sense to check if they are equal.

**Pull request merged**
